lib/p2p/Pool.ts

```diff
@@ -182,14 +183,27 @@ class Pool extends EventEmitter {
   /**
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
-  private connectNode = async ({ addresses, nodePubKey }: NodeConnectionInfo) => {
-    for (let n = 0; n < addresses.length; n += 1) {
+  private tryConnectNode = async (node: NodeInstance | NodeConnectionInfo, retryConnecting?: boolean) => {
+    const { addresses, nodePubKey } = node;
+
+    // sort by lastConnected desc
+    const sortedAddresses = [...addresses].sort((a, b) => {
+      if (!a.lastConnected) return 1;
+      if (!b.lastConnected) return -1;
+      return b.lastConnected - a.lastConnected;
+    });
+
+    for (const address of sortedAddresses) {
       try {
-        await this.addOutbound(addresses[n], nodePubKey);
-        break; // once we've successfully established an outbound connection, stop attempting new connections
-      } catch (err) {
-        this.logger.info(err);
-      }
+        await this.addOutbound(address, nodePubKey, false);
+        return; // once we've successfully established an outbound connection, stop attempting new connections
+      } catch (err) {}
```

Comment:
Should we at least log a `trace` level message here in the catch block?

Reply:
It's already being logged inside `addOutbound` or from more inner methods, which I think is the better place to do it. If we'll add logging here, it will be a duplication. 

In a case of `NODE_ALREADY_CONNECTED` it doesn't log, but I don't think it's necessary - it supposed to happen only due to CLI/gRPC call, so the thrown error will be logged.